airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestination.java

```diff
@@ -0,0 +1,85 @@
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySQLDestination extends AbstractJdbcDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySQLDestination.class);
+
+  public static final String DRIVER_CLASS = "com.mysql.cj.jdbc.Driver";
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try (final JdbcDatabase database = getDatabase(config)) {
+      String outputSchema = namingResolver.getIdentifier(config.get("database").asText());
+      attemptSQLCreateAndDropTableOperations(outputSchema, database, namingResolver, sqlOperations);
+      boolean localFileEnabled = ((MySQLSqlOperations) sqlOperations)
+          .checkIfLocalFileIsEnabled(database);
+      if (!localFileEnabled) {
+        ((MySQLSqlOperations) sqlOperations).tryEnableLocalFile(database);
+      }
+      if (!((MySQLSqlOperations) sqlOperations).isCompatibleVersion(database)) {
+        throw new RuntimeException("MySQL version not compatible with Airbyte");
+      }
+
+      return new AirbyteConnectionStatus().withStatus(Status.SUCCEEDED);
+    } catch (Exception e) {
+      LOGGER.error("Exception while checking connection: ", e);
+      return new AirbyteConnectionStatus()
+          .withStatus(Status.FAILED)
+          .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
+    }
+  }
+
+  public MySQLDestination() {
+    super(DRIVER_CLASS, new MySQLNameTransformer(), new MySQLSqlOperations());
+  }
+
+  @Override
+  protected JdbcDatabase getDatabase(JsonNode config) {
+    final JsonNode jdbcConfig = toJdbcConfig(config);
+
+    return Databases.createJdbcDatabase(
+        jdbcConfig.get("username").asText(),
+        jdbcConfig.has("password") ? jdbcConfig.get("password").asText() : null,
+        jdbcConfig.get("jdbc_url").asText(),
+        driverClass,
+        "allowLoadLocalInfile=true");
+  }
+
+  @Override
+  public JsonNode toJdbcConfig(JsonNode config) {
+    final StringBuilder jdbcUrl = new StringBuilder(String.format("jdbc:mysql://%s:%s/%s",
+        config.get("host").asText(),
+        config.get("port").asText(),
+        config.get("database").asText()));
+
+    ImmutableMap.Builder<Object, Object> configBuilder = ImmutableMap.builder()
+        .put("username", config.get("username").asText())
+        .put("jdbc_url", jdbcUrl.toString());
+
+    if (config.has("password")) {
+      configBuilder.put("password", config.get("password").asText());
+    }
+
+    return Jsons.jsonNode(configBuilder.build());
+  }
+
+  public static void main(String[] args) throws Exception {
+    final Destination destination = new MySQLDestination();
+    LOGGER.info("starting destination: {}", MySQLDestination.class);
+    new IntegrationRunner(destination).run(args);
+    LOGGER.info("completed destination: {}", MySQLDestination.class);
+  }
+}
```

Comment:
nit: missing endline of file. Running `./gradlew format` would fix formatting problems like this.